components/motor/gpio/basic.go

```diff
@@ -150,6 +153,7 @@ func (m *Motor) setPWM(ctx context.Context, powerPct float64, extra map[string]i
 		if m.PWM != nil {
 			errs = multierr.Combine(errs, m.PWM.Set(ctx, false, extra))
 		}
+		m.on = false
```

Comment:
nit: would prefer to have this next to the m.powerPct = 0.0 line for easier readability